src/parser/parser.ts

```diff
@@ -124,7 +124,7 @@ export class Parser {
     @memoize
     name(): Name | null {
         let tok = this.peek();
-        let type = get_keyword_or_name_type(this, tok.string);
+        let type = get_keyword_or_name_type(this, tok);
```

Comment:
we could also bail out early if token-type is not `NAME`

Reply:
had a look at this in 25ab2f8 - see what you think

Reply:
Yeah in lieu of having to deal with fill_token this is a good move 😄